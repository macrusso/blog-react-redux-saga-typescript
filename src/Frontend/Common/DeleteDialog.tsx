import React from 'react';
import {
  Dialog,
  DialogTitle,
  DialogActions,
  DialogContent,
  DialogContentText,
  Button,
} from '@material-ui/core';
import { selectedPost } from 'src/routes';

interface IDeleteDialog {
  object: any;
  open: boolean;
  handleCloseDialog: () => void;
  deleteFn: (id: number, postId?: number) => void;
}

const PostEditDialog: React.SFC<IDeleteDialog> = props => {
  const { object, open, handleCloseDialog, deleteFn } = props;

  const itemType = object.postId ? 'comment' : 'post';
  return (
    <Dialog open={open} onClose={handleCloseDialog}>
      <DialogTitle>{`Delete ${itemType}`}</DialogTitle>
      <DialogContent>
        <DialogContentText>
          {`Do you really want to delete this ${itemType}?`}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleCloseDialog} color="primary" variant="outlined">
          Cancel
        </Button>
        <Button
          onClick={() => {
            deleteFn(object.id, object.postId);
            handleCloseDialog();
          }}
          color="primary"
          variant="contained"
        >
          Submit
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default PostEditDialog;
