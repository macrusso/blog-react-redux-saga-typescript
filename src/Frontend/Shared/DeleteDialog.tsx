import React from "react";
import { Dialog, DialogTitle, DialogActions, DialogContent, DialogContentText, Button } from "@material-ui/core";

interface IDeleteDialog {
  object: any;
  open: boolean;
  handleCloseDialog: () => void;
  deleteFn: (id: string) => void;
}

const PostEditDialog: React.SFC<IDeleteDialog> = props => {
  const { object, open, handleCloseDialog, deleteFn } = props;

  const itemType = object.postId ? "comment" : "post";
  return (
    <Dialog open={open} onClose={handleCloseDialog}>
      <DialogTitle>{`Delete ${itemType}`}</DialogTitle>
      <DialogContent>
        <DialogContentText>{`Do you really want to delete this ${itemType}?`}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleCloseDialog} color="primary" variant="outlined">
          Cancel
        </Button>
        <Button
          onClick={() => {
            deleteFn(object._id);
            handleCloseDialog();
          }}
          color="primary"
          variant="contained"
        >
          Submit
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default PostEditDialog;
