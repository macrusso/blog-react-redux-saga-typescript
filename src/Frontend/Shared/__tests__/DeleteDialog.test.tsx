import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import DeleteDialog from "../DeleteDialog";

describe("DeleteDialog", () => {
  test("Snapshot matches", () => {
    const wrapper = shallow(
      <BrowserRouter>
        <DeleteDialog object={{}} open={true} deleteFn={jest.fn()} handleCloseDialog={jest.fn()} />
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
