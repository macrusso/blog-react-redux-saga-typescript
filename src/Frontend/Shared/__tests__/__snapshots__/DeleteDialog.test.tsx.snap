// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDialog Snapshot matches 1`] = `ShallowWrapper {}`;
